packages/react-sdk/src/components/image.tsx

```diff
@@ -1,24 +1,86 @@
 import React, { forwardRef, type ElementRef, type ComponentProps } from "react";
+import { getImageAttributes, type ImageLoader } from "../component-utils/image";
+
+const getInt = (value: unknown): number | undefined => {
+  if (typeof value === "number") {
+    return Math.round(value);
+  }
+
+  if (typeof value === "string") {
+    const vNum = Number.parseFloat(value);
+
+    if (!Number.isNaN(vNum)) {
+      return Math.round(vNum);
+    }
+  }
+
+  return undefined;
+};
+
+const DEFAULT_QUALITY = 80;
 
 const defaultTag = "img";
 
-type ImageProps = ComponentProps<typeof defaultTag>;
+type ImageProps = ComponentProps<typeof defaultTag> & {
+  quality?: number;
+  loader?: ImageLoader;
+  optimize?: boolean;
+};
 
 export const Image = forwardRef<ElementRef<typeof defaultTag>, ImageProps>(
-  (props, ref) => <img {...props} ref={ref} />
+  (
+    {
+      quality,
+      loader,
+      optimize,
+      width: widthProp,
+      height: heightProp,
+      ...imageProps
+    },
+    ref
+  ) => {
+    const width = getInt(widthProp);
+    const height = getInt(heightProp);
+
+    // Temporary set to false, to support previous image behaviour
+    const DEFAULT_OPTIMIZE = false;
+
+    const imageAttributes = getImageAttributes({
+      src: imageProps.src,
+      srcSet: imageProps.srcSet,
+      sizes: imageProps.sizes,
+      width,
+      height,
+      quality: Math.max(Math.min(getInt(quality) ?? DEFAULT_QUALITY, 100), 0),
+      loader,
+      optimize: optimize ?? DEFAULT_OPTIMIZE,
+    }) ?? { src: imagePlaceholderSvg };
```

Comment:
I would creat one function in component-utils that takes the props and does all of this

Reply:
`ComponentProps<typeof defaultTag>` is react specific i.e event handlers like onLoad etc.
So we can't use it as function arg type. 
Also I would prefer to always have the most narrow type for most functions.

In ideal world `width`, `height` would be already validated `int | undefined` from data, 
so I leaved them here.

As no reason to have them as strings in a system if spec says that it's integer numbers.

Reply:
you can type the props you need in that function, don't understand the prob